src/components/AccordionItem/AccordionItem.js

```diff
@@ -84,11 +84,13 @@ export default class AccordionItem extends Component {
     this.props.onHeadingClick({ isOpen: open, event: evt });
   };
 
-  handleKeyPress = evt => {
-    const isKeyPressTarget = evt.target === evt.currentTarget;
-    const isValidKeyPress = [13, 32].indexOf(evt.which) !== -1;
-
-    if (isKeyPressTarget && isValidKeyPress) {
+  handleKeyDown = evt => {
+    // Esc key
+    if (
+      evt.which === 27 &&
```

Comment:
Could use the helpers in: https://github.com/IBM/carbon-components-react/blob/master/src/tools/key.js if they'd be helpful, e.g.

```js
if (match(evt, keys.ESC)) {
    // ...
}
```

Reply:
wasn't aware we had these helpers, thanks!